cityofaustin/atd-moped
# Update to graphql-engine 2.7.0+

This PR began as a lighthearted conversation about what tech debt we might be able to address with this, um, conspicuous PR number. 

## TLDR

This PR represents multiple things:

1) Changing our local development environments to use the latest `graphql-engine` from Hasura
1) Committing to move forward to roll out the same version in `test`, `staging`, and `production` as soon as is reasonably prudent & possible

⚠️ Please withhold your approval until you are comfortable with both of these goals.

## Approval process

Because of the reaching effects of such a change, I intend to seek more approvals than normal, with a goal of getting sign-offs from the bulk of the development team. 

## Running `graphql-engine` 2.7+ locally prior to this being merged

Developers can use the latest `graphql-engine` during other work if they want. This can be accomplished by making [this line](https://github.com/cityofaustin/atd-moped/blob/main/moped-database/docker-compose.x86_64.yml#L20) look like [this line](https://github.com/cityofaustin/atd-moped/blob/the-devil-is-in-the-details/moped-database/docker-compose.yml#L6) and restarting the DB cluster. 

## General notes

This PR intends to close https://github.com/cityofaustin/atd-data-tech/issues/9385. 

## Special testing considerations

Please read, at the very least, each of the "breaking changes" enumerated by Hasura [here](https://hasura.io/docs/latest/graphql/core/guides/upgrade-hasura-v2/#hasura-v2-behaviour-changes). Carefully consider each one in the scope of the entire Moped stack, and become comfortable that each either:

1) Does not have potential to be breaking to our application stack, or
1) Does have potential to be breaking to our stack, but that the change is found to be non-breaking in our particular case.

Additional consideration of the behavior and breaking changes in the subsequent minor releases to `graphql-engine` would be helpful as well.

## Testing
Testing on `test` should work as normal, rolling out `2.7.0`, and this is important to do. 

Also equally important, testing on our local development environments, is needed. Please: 

1) Spin up the DB from this branch.
1) Spin up the app and try any which way you can to break it. Testing anything and everything is fair game.
1) Use it while you do your normal development process as you're able.



### Particular code to consider and test

- [ ] Our `GQLAbstract` library takes a configuration which it uses to build a graphql query. I believe we have two distinct configurations which we feed into the library, and they can be found [here](https://github.com/cityofaustin/atd-moped/blob/the-devil-is-in-the-details/moped-editor/src/views/projects/projectsListView/ProjectsListViewQueryConf.js) and [here](https://github.com/cityofaustin/atd-moped/blob/the-devil-is-in-the-details/moped-editor/src/views/staff/StaffListViewQueryConf.js). Do you agree that these should work and do work?

This PR contains comments in the code that helped me answer the question above. I intend to take those comments back out of this PR prior to merging. They are only helpful during this PR evaluation, I believe.

---
#### Ship list
- [x] Application _passes_ the [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#)
- [ ] Comfort with activity log functionality
